hal/inc/spi_hal.h

```diff
@@ -55,6 +55,8 @@ typedef enum hal_spi_state_t {
 
 typedef void (*hal_spi_dma_user_callback)(void);
 typedef void (*hal_spi_select_user_callback)(uint8_t);
+typedef hal_spi_dma_user_callback HAL_SPI_DMA_UserCallback; // For backwards compatibility only
```

Comment:
These can now also be moved into compat header. We can also put deprecated attribute on them.